src/create-catalog-v001.xml.py

```diff
@@ -0,0 +1,366 @@
+from xml.dom import minidom
+
+from rdflib import OWL, RDF, Graph, URIRef
+
+NS_OWL = OWL
+NS_RDF = RDF
+
+
+# XML prolog, as generated by Protégé.
+XML_VERSION_INFO = '<?xml version="1.0" encoding="UTF-8" standalone="no"?>'
+
+
+def main() -> None:
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--debug", action="store_true")
+    parser.add_argument(
+        "--catalog-xml",
+        help="A generated catalog-v001.xml file for some dependent or imported ontology.  This should be supplied for ontologies not covered by the dependency_files_tsv argument, nor meant to be crawled upon using domain_directories_tsv.  (For instance, this could be used to import relative file references for ontologies tracked as Git submodules.)",
+        action="append",
+    )
+    # "x" mode - exclusive creation.
+    # https://docs.python.org/3/library/functions.html#open
+    parser.add_argument("out_xml", type=argparse.FileType("x"))
+    parser.add_argument(
+        "domain_directories_tsv",
+        help="A two-column file, with column 1 being a string prefix in-common to ontology prefix IRIs, and column 2 being a file system directory relative to top_srcdir that is the root directory housing that ontology's files.  Directories specified in this file will be recursively walked to discover ontology graph files.  This file may be empty, but it must exist.",
+        type=argparse.FileType("r"),
+    )
+    parser.add_argument(
+        "dependency_files_tsv",
+        help="A two-column file, with column 1 being an ontology reference IRI (ontology IRI or version IRI), and column 2 being a local, version-controlled file relative to top_srcdir that houses the corresponding ontology data.  This file may be empty, but it must exist.",
+        type=argparse.FileType("r"),
+    )
+    parser.add_argument("top_srcdir", help="The root directory of the Git repository for this ontology.  In the two TSV arguments, the variable top_srcdir is substituted with this value.")
+    parser.add_argument(
+        "in_ttl",
+        help="Input graph files.  Due to the target use case of Protégé Catalog files, these are required to be in the same directory.",
+        nargs="+",
+    )
+    args = parser.parse_args()
+
+    logging.basicConfig(level=logging.DEBUG if args.debug else logging.INFO)
+
+    top_srcdir_abspath = Path(args.top_srcdir).resolve()
+    if not top_srcdir_abspath.exists():
+        raise FileNotFoundError(args.top_srcdir)
+    if not top_srcdir_abspath.is_dir():
+        raise NotADirectoryError(args.top_srcdir)
+    logging.debug("top_srcdir_abspath = %r.", top_srcdir_abspath)
+
+    focus_graph = Graph()
+
+    focus_graph_srcdir_abspaths: Set[Path] = set()
+    for in_ttl in args.in_ttl:
+        focus_graph_abspath = Path(in_ttl).resolve()
+        focus_graph.parse(str(focus_graph_abspath))
+        focus_graph_srcdir_abspaths.add(focus_graph_abspath.parent)
+    if len(focus_graph_srcdir_abspaths) > 1:
```

Comment:
There's no `else` clause catching an empty graph. Is that a non-fatal error?

Reply:
I don't think there's a test written for that case.  `args.in_ttl` is required to have at least one graph passed it, but you're right, the graph could be syntactically valid and have 0 triples in it.

However, if it is empty, then the constructed transitive closure would be empty, making the `catalog-v001.xml` file a "base case" form - no redirects defined.  This seems like a happy-enough outcome to me.

Do you need this tested before you approve?  If so, we'll need to make, test, and merge a follow-on PR for Issue 449 before merging PR 530.